### ISSUE TITLE
gradient accumulation for very large embeddings

### ISSUE THREAD

**Issue by AdrianYu** [tensorflow/tensorflow#31637]
**System information**
- TensorFlow version (you are using): 1.12
- Are you willing to contribute it (Yes/No): yes, but don't know how.

**Describe the feature and the current behavior/state.**

**Will this change the current api? How?**
It certainly will add some new apis. Currently, in a distributed environment, we can use `tf.contrib.opt.AGNOptimizer` class to do gradient accumulation. However this optimizer converts all gradients (Tensor or IndexedSlices) to dense types which consumes quite large memory. In certain scenarios, say we've got some very large embedding variables which can't fit into one single machine, this would cause OOM on workers.

**Who will benefit with this feature?**
Anyone who has some very large variables in their models, and want to use gradient accumulation to accelerate training.

**Any Other info.**
None.

**Comment by ravikyram** [comment]
Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by jvishnuvardhan** [comment]
@AdrianYu Can you try recent TF versions and see whether the OOM issues persists. There were lot of improvements more recently. Please let us know what you think? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
